src/deadline/job_attachments/api/manifest.py

```diff
@@ -247,3 +260,182 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
+    attachments: dict = job["attachments"]
+    input_manifest_paths: List[Tuple[str, str]] = [
+        (manifest["inputManifestPath"], manifest["rootPath"])
+        for manifest in attachments["manifests"]
+    ]
+
+    # Download each input_manifest_path
+    for input_manifest in input_manifest_paths:
+        asset_manifest: BaseAssetManifest = get_manifest_from_s3(
+            manifest_key=(s3_prefix / input_manifest[0]).as_posix(),
+            s3_bucket=queue_s3_settings.s3BucketName,
+            session=queue_role_session,
+        )
+        if asset_manifest is not None:
+            logger.echo(f"Downloaded input manifest for root: {input_manifest[1]}")
+            add_manifest_by_root(
+                manifests_by_root=manifests_by_root, root=input_manifest[1], manifest=asset_manifest
+            )
+
+    # Now handle step-step dependencies
+    if step_id is not None:
+        # Get Step-Step dependencies.
+        nextToken = ""
+        step_dep_response = deadline.list_step_dependencies(
+            farmId=farm_id,
+            queueId=queue_id,
+            jobId=job_id,
+            stepId=step_id,
+            nextToken=nextToken,
+        )
+        for step in step_dep_response["dependencies"]:
+            logger.echo(f"Found Step-Step dependency. {step['stepId']}")
+
+        # Get manifests for the step-step dependency
+        step_manifests_by_root: Dict[str, List[BaseAssetManifest]] = (
+            get_output_manifests_by_asset_root(
+                queue_s3_settings,
+                farm_id,
+                queue_id,
+                job_id,
+                step_id=step_id,
+                session=queue_role_session,
+            )
+        )
+        # Merge all manifests by root.
+        for root in step_manifests_by_root.keys():
+            for manifest in step_manifests_by_root[root]:
+                logger.echo(f"Found step-step output manifest for root: {root}")
+                add_manifest_by_root(
+                    manifests_by_root=manifests_by_root, root=root, manifest=manifest
+                )
+
+    # Finally, merge all manifest paths to create unified manifests.
+    # TODO: Filter outputs by path
+    merged_manifests: Dict[str, BaseAssetManifest] = {}
+    for root in manifests_by_root.keys():
+        merged_manifest = merge_asset_manifests(manifests_by_root[root])
+        if merged_manifest:
+            merged_manifests[root] = merged_manifest
+
+    # Save the manifest files to disk.
+    for root in merged_manifests.keys():
+        # Save the merged manifest as {root}_{hash}_timestamp.
+        root_hash: str = hash_data(
+            root.encode("utf-8"), merged_manifests[root].get_default_hash_alg()
+        )
+        timestamp = datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S")
+        manifest_name = root.replace("/", "_")
+        manifest_name = manifest_name[1:] if manifest_name[0] == "_" else manifest_name
```

Comment:
nit: can use built-in str function for this too
```suggestion
        manifest_name = manifest_name.removeprefix("_")
```

Reply:
Done!

Reply:
Cannot - this is not Py 3.8 compatible.